### PR TITLE
Decorate Person Class

### PR DESCRIPTION
**Pull request by studiosnabeel**
pull request
### In This Pull Request:🚀🎉💫
- Add decorate to Person class 🌐

### CODE REVIEW

**Comment by prolajumokeoni** [comment]
Hi @studiosnabeel 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that does not have the key features section in your readme

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: decorator.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require_relative 'nameable'
+
+class Decorator < Nameable
```
Comment (zieeco) [inline decorator.rb:3]: - Hey, we also need to access the nameable by adding the `attr_accessor :nameable_obj`. please, kindly add it to enable the read and write access.

File: README.md [hunk 2]
```diff
@@ -1,135 +1,150 @@
...
+I would like to thank...
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- FAQ (optional) -->
+
+## ❓ FAQ <a name="faq"></a>
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
```
Comment (zieeco) [inline README.md]: - Hey, since this is not in the readme, kindly remove it to make your readme more professional.

File: README.md [hunk 3]
```diff
@@ -1,135 +1,150 @@
...
+If you like this project...
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- ACKNOWLEDGEMENTS -->
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+> Give credit to everyone who inspired your codebase.
```
Comment (zieeco) [inline README.md]: - Please, let's kindly remove all this placeholder text as it's there to guide you on how to update the content on your readme, so it shouldn't be in the readme after you've made an update to the readme to suit your project needs.

File: README.md [hunk 4]
```diff
@@ -1,135 +1,150 @@
...
+
+Feel free to check the [issues page](https://github.com/studiosnabeel/ruby-school-library/issues).
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- SUPPORT -->
+
+## ⭐️ Show your support <a name="support"></a>
+
+> Write a message to encourage readers to support your project
```
Comment (zieeco) [inline README.md]: - Please, let's kindly remove all this placeholder text as it's there to guide you on how to update the content on your readme, so it shouldn't be in the readme after you've made an update to the readme to suit your project needs.

File: README.md [hunk 5]
```diff
@@ -1,135 +1,150 @@
...
+
+👤 **Nabeel Akbar**
+
+- GitHub: [@studiosnabeel](https://github.com/studiosnabeel)
+- Twitter: [@studiosNabeel](https://twitter.com/StudiosNabeel)
+- LinkedIn: [@nabeel-akbar](https://www.linkedin.com/in/studiosNabeel/)
+
+## 🔭 Future Features <a name="future-features"></a>
+
+> Describe 1 - 3 features you will add to the project.
```
Comment (zieeco) [inline README.md]: - Please, let's kindly remove all this placeholder text as it's there to guide you on how to update the content on your readme, so it shouldn't be in the readme after you've made an update to the readme to suit your project needs.

File: README.md [hunk 6]
```diff
@@ -1,135 +1,150 @@
...
+    - [Key Features](#key-features)
+- [💻 Getting Started](#getting-started)
+  - [Setup](#setup)
+  - [Prerequisites](#prerequisites)
+- [👥 Authors](#authors)
+- [🔭 Future Features](#future-features)
+- [🤝 Contributing](#contributing)
+- [⭐️ Show your support](#support)
+- [🙏 Acknowledgements](#acknowledgements)
+- [❓ FAQ](#faq)
```
Comment (zieeco) [inline README.md]: - Hey, since this is not in the readme, kindly remove it to make your readme more professional.

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @studiosnabeel ,


Great job so far! 👏 
You've done well, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice class decoration. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._





## Optional suggestions
- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Munsa1 — APPROVED** [review]
Hi @studiosnabeel, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Made requested changes by previous reviewer :heavy_check_mark: 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**